torch/fx/experimental/proxy_tensor.py

```diff
@@ -480,57 +559,38 @@ def __contains__(self, key: py_sym_types):
-        return attr_val
-
-    def create_arg(self, a: Any):
-        if isinstance(a, torch.nn.Parameter):
-            for n, p in self.root.named_parameters():
-                if a is p:
-                    return self.create_node('get_attr', n, (), {})
-            qualname: Optional[str] = None
-
-            if not qualname:
-                i = 0
-                while True:
-                    qualname = f'_param_constant{i}'
-                    if not hasattr(self.root, qualname):
-                        break
-                    i += 1
-                setattr(self.root, qualname, a)
-
-            return self.create_node('get_attr', qualname, (), {})
-        elif isinstance(a, (SymInt, SymFloat, SymBool)):
-            assert a.node.constant is not None
-            return a.node.constant
-        return super().create_arg(a)
-
-    def unwrap_proxy(self, e):
-        if isinstance(e, torch.Tensor):
-            return get_proxy_slot(e, self, e, lambda e: e.proxy)
-        elif isinstance(e, (torch.SymInt, torch.SymFloat, torch.SymBool)):
-            return get_proxy_slot(e, self, e, lambda e: e())
-        elif isinstance(e, torch.ScriptObject):
-            return get_proxy_slot(e, self, e)
-        else:
-            return e
+class ProxyTorchFunctionMode(TorchFunctionMode):
+
+    metadata_fns = [
+        torch.Tensor.detach,  # type: ignore[has-type]
+        torch.ops.prim.device.default,  # type: ignore[has-type]
+        torch.Tensor.is_quantized.__get__,  # type: ignore[has-type]
+        torch.Tensor.stride,  # type: ignore[has-type]
+        torch.Tensor.storage_offset,  # type: ignore[has-type]
+        torch.ops.aten.is_contiguous.memory_format,  # type: ignore[has-type]
+        torch.ops.aten.is_strides_like_format.default,  # type: ignore[has-type]
+        torch.ops.aten.is_non_overlapping_and_dense.default,  # type: ignore[has-type]
+        torch.ops.aten.size.default,  # type: ignore[has-type]
+        torch.ops.aten.sym_size.default,  # type: ignore[has-type]
+        torch.ops.aten.stride.default,  # type: ignore[has-type]
+        torch.ops.aten.sym_stride.default,  # type: ignore[has-type]
+        torch.ops.aten.storage_offset.default,  # type: ignore[has-type]
+        torch.ops.aten.sym_storage_offset.default,  # type: ignore[has-type]
+        torch.ops.aten.numel.default,  # type: ignore[has-type]
+        torch.ops.aten.sym_numel.default,  # type: ignore[has-type]
+        torch.ops.aten.dim.default,  # type: ignore[has-type]
+    ]
+
+    def __init__(self, tracer: PythonKeyTracer):
+        super().__init__()
+        self.tracer = tracer
 
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs or {}
+        if func in torch_function_passthrough or func in all_prims() or func in ProxyTorchFunctionMode.metadata_fns:
+            return func(*args, **kwargs)
+        self.tracer.traced_func = [(func.__name__, func)]
```

Comment:
The first value in the tuple is supposed to differentiate between two separate calls to the same function which are back to back -- you can modify your test case to call linear twice in a row and see what happens.
I don't have a good idea for how to do this though, but wondering if you have any ideas.